pkg/service/restore_memory.go

```diff
@@ -198,6 +199,29 @@ func (r *RestoreMemory) findIncrementalBackupsForNamespace(
 	return filteredIncrementalBackups, nil
 }
 
+func (r *RestoreMemory) RestoreConfiguration(routine string, toTimeMillis int64) ([]byte, error) {
+	backend, found := r.backends[routine]
+	if !found {
+		return nil, fmt.Errorf("backend '%s' not found for restore", routine)
+	}
+	fullBackups, err := r.findLastFullBackup(backend, toTimeMillis)
+	if err != nil {
+		return nil, fmt.Errorf("last full backup not found: %v", err)
+	}
+
+	// fullBackups has backups for multiple namespaces, but same timestamp, they share same configuration.
+	lastFullBackup := fullBackups[0]
```

Comment:
Check `fullBackups` is not empty.

Reply:
findLastFullBackup will return error if nothing is found, but I'll add check anyway.